classlib/src/main/java/org/teavm/classlib/java/util/TArrays.java

```diff
@@ -1463,10 +1458,9 @@ public static int hashCode(float[] a) {
         if (a == null) {
             return 0;
         }
-        int hash = 0xA5A537FC;
+        int hash = 1;
         for (int i = 0; i < a.length; ++i) {
-            int h = TFloat.floatToIntBits(a[i]);
-            hash = TInteger.rotateLeft(h, 4) ^ TInteger.rotateRight(h, 7) ^ TInteger.rotateLeft(hash, 13);
+            hash = 31 * hash + Float.hashCode(a[i]);
```

Comment:
What's the purpose of changes like this?

Reply:
If you check the documentation of Arrays, you will see that Arrays.hashCode should be equal to hashcode of equal list.